accepted/00076-live-patching-eol-data.md

```diff
@@ -0,0 +1,119 @@
+
+# Detailed design
+[design]: #detailed-design
+
+## Live patching EOL information
+
+SUSE provides lifecycle information on any product to users via 'zypper lifecycle' plugin. The actual data about a product is provided in a plain CSV file in a supplementary package per product. In case of live patching, this package is called `lifecycle-data-sle-live-patching` and is installed together with the product. Zypper lifecycle plugin reads this data to display human-friendly information to users.
+
+Uyuni shall use this data to provide the following:
+ - Display EOL dates for the current live patch in system details
+ - Display an out-of-support systems list in the home page
+ - Use the notifications system and/or send emails to report live patched systems approaching/past EOL
+ - The notifications/emails will be sent in advance according to a time value configurable from the UI
+
+## Data Retrieval
+
+The lifecycle information for a product is provided with `*.lifecycle` files inside a separate package. Uyuni shall unpack this RPM and read the data from the extracted CSV file during reposync of this product. The data shall be stored in the database per package.
+
+An example SQL query to retrieve the list of RPMs to unpack and the corresponding files to read from:
+
+```sql
+SELECT
+    c.name filename,
+    p.path rpm_path
+FROM
+    rhnPackage p
+    JOIN rhnPackageFile f ON p.id = f.package_id
+    JOIN rhnPackageCapability c ON c.id = f.capability_id
+    JOIN rhnPackageEvr e ON p.evr_id = e.id
+WHERE
+    --Select packages that have '*.lifecycle' files in them
+    c.name LIKE '%.lifecycle' AND
+    e.evr = (
+        --Select latest EVR for a package name
+        SELECT MAX(evr)
+        FROM rhnPackageEvr se
+            JOIN rhnPackage sp ON se.id = sp.evr_id
+        WHERE sp.name_id = p.name_id
+    );
+```
+
+The output is a list of RPM paths in the local filesystem and the packed filenames to extract the data from:
+```
+                      filename                       |                         rpm_path
+-----------------------------------------------------+----------------------------------------------------------
+ /var/lib/lifecycle/data/sle-live-patching.lifecycle | .../lifecycle-data-sle-live-patching-1-10.79.1.noarch.rpm
+
+```
+
+The list of RPMs are unpacked into a temp directory and the corresponding files are read into a table called `susePackageLifecycle` with the following structure:
+ - `package_id` as a foreign key to `rhnPackage` table
+ - `eol_date` of `DATE` type
+
+ The data stored in this table is available to be queried and consumed in the UI.
+
+ *For other methods of data retrieval that are evaluated in the design, see [Alternatives](#alternatives).*
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+ - Unpacking must be done during reposync, adding complication
```

Comment:
An option would be to do it when we generate the metadata for a new synced channel.
Of cause we need to call an external program from the taskomatic job.

But during reposync is also ok.

Reply:
I think when implemented properly with enough logging, reposync is the most natural choice.